aziot/src/check.rs

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::collections::BTreeMap;
+use std::io::prelude::*;
+use std::str::FromStr;
+
+use anyhow::Result;
+use colored::Colorize;
+use structopt::StructOpt;
+
+use aziot_check_common::{
+    CheckOutputSerializable, CheckOutputSerializableStreaming, CheckResultSerializable,
+    CheckResultsSerializable,
+};
+
+use crate::internal::check::{
+    AdditionalInfo, CheckResult, CheckerCache, CheckerCfg, CheckerShared,
+};
+
+#[derive(StructOpt)]
+#[structopt(about = "Check for common config and deployment issues")]
+pub struct CheckCfg {
+    /// Space-separated list of check IDs. The checks listed here will not be run.
+    /// See 'aziot check-list' for details of all checks.
+    #[structopt(
+        long,
+        value_name = "DONT_RUN",
+        value_delimiter = " ",
+        use_delimiter = true
+    )]
+    dont_run: Vec<String>,
+
+    /// Output format. Note that JSON output contains some additional information
+    /// like OS name, OS version, disk space, etc.
```

Comment:
Add `One of "text" or "json".` between the two sentences?

(`"json-stream" omitted intentionally.)